### PR TITLE
Add tests that cover msan explicit_bzero issue

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

Follow-up for #7301: add a test case with uninitialised memory that fails under msan if `mbedtls_platform_zeroize` is optimised out, or if msan can't correctly handle it.

On recent Ubuntu, this fails until #7301 is merged.

## Gatekeeper checklist

- [x] **changelog** not required
- [x] **backport** not required - not in 2.28
- [x] **tests** provided

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
> This should fail the CI

I don't think so? https://github.com/Mbed-TLS/mbedtls/pull/7301 doesn't fix a CI failure. Only if you run the Msan job on a more modern system.

**Comment by daverodgman** [comment]
> > This should fail the CI
> 
> I don't think so? #7301 doesn't fix a CI failure. Only if you run the Msan job on a more modern system.

Ah, sorry, yes of course. I've tested locally and it does fail until 7301 is present, and then passes.

File: tests/suites/test_suite_platform_util.function [hunk 1]
```diff
@@ -39,3 +39,13 @@ void mbedtls_platform_zeroize(int len, int null)
...
+
+/* BEGIN_CASE */
+void mbedtls_platform_zeroize_uninitialised(int len, int p)
+{
+    char buf[128];
+    mbedtls_platform_zeroize(buf, len);
+
+    TEST_EQUAL(buf[p], 0);
+}
+/* END_CASE */
```
Comment (tom-cosgrove-arm) [inline tests/suites/test_suite_platform_util.function:51]: (minor) missing newline at EOF

**Comment by tom-cosgrove-arm** [comment]
Should we have some comment to explain what we're testing?

**Comment by daverodgman** [comment]
> Should we have some comment to explain what we're testing? (Within the committed files, that is!)

I could add something if you like - but the commit history is going to lead back here so not sure it adds very much.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by paul-elliott-arm — APPROVED** [review]

**Pull request merged**